Compile error when uses with hmac 0.9

Code:

```rust
use hmac::{Hmac, NewMac};
use jwt::SignWithKey;
use sha2::Sha256;
use std::collections::BTreeMap;
use jwt::algorithm::rust_crypto;

fn main() {
    let key: Hmac<Sha256> = Hmac::new_varkey(b"some-secret").unwrap();
    let mut claims = BTreeMap::new();
    claims.insert("sub", "someone");
    let token_str = claims.sign_with_key(&key).unwrap();
    assert_eq!(token_str, "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJzb21lb25lIn0.5wwE1sBrs-vftww_BGIuTVDeHtc1Jsjo-fiHhDwR8m0");
}
```

Error:

```
error[E0277]: the trait bound `Hmac<Sha256>: AsRef<(dyn SigningAlgorithm + 'static)>` is not satisfied
  --> src/test.rs:11:42
   |
11 |     let token_str = claims.sign_with_key(&key).unwrap();
   |                                          ^^^^ the trait `AsRef<(dyn SigningAlgorithm + 'static)>` is not implemented for `Hmac<Sha256>`
   |
   = note: required because of the requirements on the impl of `SigningAlgorithm` for `Hmac<Sha256>`
```

This can be fixed by downgrading hmac to 0.8.
Maybe we can add a note some where so others won't spend hours trying to find out why the example code doesn't work.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,9 +16,9 @@ features = ["openssl"]
 
 [dependencies]
 base64 = "0.12"
-crypto-mac = "0.8"
+crypto-mac = "0.9"
 digest = "0.9"
-hmac = "0.8"
+hmac = "0.9"
 sha2 = "0.9"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"

